### PR TITLE
Added mongodb

### PR DESCRIPTION
**Pull request by realharley91**

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
+  const host_object = await db.collection('hosts').findOne({ host })
+
+  return host_object ? host_object.pathname : []
 };
 
-const count = (url) => {
-  return get(url) ? get(url).length : 0;
+const count = async (url) => {
+  const data = await get(url)
+  console.log(data)
```
Comment (cagataycali) [inline db.js:35]: We have to remove this line

Comment (realharley91) [reply db.js:35]: Sorry my bad, will remove it along with the yarn-lock file

File: db.js [hunk 2]
```diff
@@ -1,40 +1,45 @@
...
-      db[host][pathname] = [];
-      db[host][pathname].push(data);
-    }
-  } else {
-    db[host] = {};
-    db[host][pathname] = [];
-    db[host][pathname].push(data);
-  }
+
+  const db = await connect()
```
Comment (cagataycali) [inline db.js:14]: We have to move the connection top of file

File: db.js [hunk 3]
```diff
@@ -1,40 +1,45 @@
...
+  const data = await get(url)
+  console.log(data)
+  return data ? data.length : 0
 };
 
-const all = () => db;
+const all = async () => {
+  const db = await connect()
+  const hosts = await db.collection('hosts').find({})
+  return hosts
```
Comment (cagataycali) [inline db.js:42]: if you return the data w/o manipulation. This method should look like this:

```javascript
const all = async () => db.collection('hosts').find({})
```

File: index.js [hunk 4]
```diff
@@ -1,29 +1,41 @@
...
+// configure app
+const app = express()
 app.use(cors());
 app.use(express.json())
 app.use(express.static("docs"));
 
-app.get("/", (req, res) => res.json(all()));
+// routes
+app.get("/", async(req, res) => {
+  res.json(await all())
```
Comment (cagataycali) [inline index.js:18]: What if rejects?

File: index.js [hunk 5]
```diff
@@ -1,29 +1,41 @@
...
+});
 
-app.get("/a/*", (req, res) => {
-  const data = get(getHREF(req));
+app.get("/a/*", async (req, res) => {
+  const data = await get(getHREF(req));
```
Comment (cagataycali) [inline index.js:22]: what if rejects?

File: index.js [hunk 6]
```diff
@@ -1,29 +1,41 @@
...
-app.get("/c/*", (req, res) => res.json({count: count(getHREF(req))}));
+app.get("/c/*", async (req, res) => {
+  res.json({count: await count(getHREF(req))})
+});
 
-app.post('/', (req, res) => {
+app.post('/', async (req, res) => {
   // req.body must be includes {href: 'http://localhost:3000'} like object.
-  add(req.body);
+  await add(req.body);
```
Comment (cagataycali) [inline index.js:35]: Why we're blocking the response?

File: package.json [hunk 7]
```diff
@@ -4,7 +4,8 @@
   "description": "We want create new-gen analytics platform for developers to developers. This project aim is one click analytics solution for open-source projects & more.",
   "main": "index.js",
   "scripts": {
-    "start": "node index.js"
+    "start": "node index.js",
+    "dev": "nodemon index.js"
```
Comment (cagataycali) [inline package.json:8]: we do not need nodemon at all

File: package.json [hunk 8]
```diff
@@ -21,6 +22,11 @@
   "homepage": "https://github.com/cagataycali/analytics#readme",
   "dependencies": {
     "cors": "^2.8.5",
-    "express": "^4.17.1"
+    "dotenv": "^8.2.0",
```
Comment (cagataycali) [inline package.json:25]: we don't need dotenv at all.

**Comment by cagataycali** [comment]
We can fix tho, @barisesen

**Pull request merged**